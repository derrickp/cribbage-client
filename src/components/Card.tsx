
import * as React from "react";

import * as classNames from "classnames";
import { Draggable } from 'react-beautiful-dnd';

import { Card as DataCard } from "../data/Card";

export interface CardProps {
    card: DataCard;
}

export class Card extends React.Component<CardProps, any> {

    constructor(props: CardProps) {
        super(props);

        this.getCardClass = this.getCardClass.bind(this);
    }

    getCardClass(dragging: boolean): string {
        const drag = ({
            [CardClasses.DRAGGING]: dragging
        } as any);
        return classNames(CardClasses.CARD, drag);
    }

    render() {
        const card = this.props.card;
        return <Draggable key={card.fullName} draggableId={card.fullName}>
            {(provided: any, snapshot) => (
                <div>
                    <div
                        ref={provided.innerRef}
                        style={{...provided.draggableStyle}}
                        className={this.getCardClass(snapshot.isDragging)}
                        {...provided.dragHandleProps}
                    >
                        {card.fullName}
                    </div>
                    {provided.placeholder}
                </div>
            )}
        </Draggable>
    }
}

export enum CardClasses {
    CARD = "card",
    DRAGGING = "card-dragging"
}