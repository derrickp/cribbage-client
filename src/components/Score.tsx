
import * as React from "react";

export interface ScoreProps {
    score: number;
}

export function Score(props: ScoreProps) {
    return <div className={Classes.SCORE}>{`Your current score is ${props.score}`}</div>;
}

export enum Classes {
    SCORE = "score"
}