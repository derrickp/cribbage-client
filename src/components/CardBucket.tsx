
import * as React from "react";

import * as classNames from "classnames";
import { Droppable } from 'react-beautiful-dnd';

import { Card } from "../data/Card";
import { Card as CardComponent } from "./Card";

export interface CardBucketProps {
    id: string;
    name: string;
    cards: Card[];
    maxSize?: number;
}

export interface CardBucketState { }

export class CardBucket extends React.Component<CardBucketProps, CardBucketState> {

    constructor(props: CardBucketProps) {
        super(props);
        this.getBucketClass = this.getBucketClass.bind(this);
    }

    getBucketClass(dragging: boolean): string {
        
        const drag = ({
            [BucketClasses.DRAGGING]: dragging
        } as any);
        return classNames(BucketClasses.BUCKET, drag);
    }

    render() {
        const cards = this.props.cards.map(card => <CardComponent key={card.key} card={card} />)
        return (
            <div>
                {this.props.name}
                <Droppable droppableId={this.props.id}>
                {(provided, snapshot) => (
                    <div
                        ref={provided.innerRef}
                        className={this.getBucketClass(snapshot.isDraggingOver)}
                    >
                        {cards}
                        {provided.placeholder}
                    </div>
                )}
            </Droppable>
            </div>
        )
    }
}

export enum BucketClasses {
    BUCKET = "bucket",
    DRAGGING = "dragging"
};