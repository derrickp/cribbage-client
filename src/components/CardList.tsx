
import * as React from "react";
import * as classNames from "classnames";

import { Droppable } from 'react-beautiful-dnd';
import Select from "react-select";

import { Card } from "../data/Card";
import { Card as CardComponent } from "./Card";
import { Option } from "../data/Option";

export interface CardListProps {
    id: string;
    name: string;
    cards: Card[];
    sortCards?: (sortOption: Option, id: string) => void;
    sortOptions?: Option[];
    sortValue?: Option;
    sortEnabled?: boolean;
    additionalClass?: string;
}

interface CardListState {
    filter: string;
}

export class CardList extends React.Component<CardListProps, CardListState> {

    constructor(props: CardListProps) {
        super(props);
        this.state = {
            filter: ""
        }
        this.getBucketClass = this.getBucketClass.bind(this);
        this.sortChanged = this.sortChanged.bind(this);
    }

    getBucketClass(dragging: boolean): string {
        const classMap = ({
            [Classes.DRAGGING]: dragging
        } as any);
        return classNames(Classes.LIST, classMap, this.props.additionalClass);
    }

    sortChanged(option: Option) {
        this.props.sortCards(option, this.props.id);
    }

    render() {
        const cards = this.props.cards.map(card => <CardComponent key={card.key} card={card} />);
        const sortValue = this.props.sortValue ? this.props.sortValue.value : undefined;

        return (
            <Droppable droppableId={this.props.id}>
                {(provided, snapshot) => (
                    <div className={Classes.LIST_CONTAINER}>
                        <div className="header">
                            <h4>
                                {this.props.name}
                            </h4>
                            {this.props.sortEnabled && <Select options={this.props.sortOptions} value={sortValue} placeholder="Sort by" clearable={false} onChange={this.sortChanged}></Select>}
                        </div>
                        <div
                            ref={provided.innerRef}
                            className={this.getBucketClass(snapshot.isDraggingOver)}>
                            {cards}
                            {provided.placeholder}
                        </div>
                    </div>
                )}
            </Droppable>
        );
    }
}

export enum Classes {
    LIST = "card-list",
    LIST_CONTAINER = "list-container",
    DRAGGING = "drag-over"
};