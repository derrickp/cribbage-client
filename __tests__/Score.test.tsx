
import * as React from "react";

import { create } from "react-test-renderer";

import { Score } from "../src/components/Score";

describe("Score", () => {
    it("renders the score", () => {
        const component = create(<Score score={2} />).toJSON();
        expect(component).not.toBe(undefined);
        expect(component).toMatchSnapshot();
    });
});