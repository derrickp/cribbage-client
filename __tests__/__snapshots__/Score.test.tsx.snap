// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders the score 1`] = `
<div
  className="score"
>
  Your current score is 2
</div>
`;
